admin/server/users.go

```diff
@@ -114,6 +114,51 @@ func (s *Server) GetCurrentUser(ctx context.Context, req *adminv1.GetCurrentUser
 
 	return &adminv1.GetCurrentUserResponse{
 		User: userToPB(u),
+		Preferences: &adminv1.UserPreferences{
+			Timezone: &u.PreferenceTimezone,
+		},
+	}, nil
+}
+
+func (s *Server) UpdateUserPreferences(ctx context.Context, req *adminv1.UpdateUserPreferencesRequest) (*adminv1.UpdateUserPreferencesResponse, error) {
+	// Return an empty result if not authenticated.
+	claims := auth.GetClaims(ctx)
+	if claims.OwnerType() == auth.OwnerTypeAnon {
+		return &adminv1.UpdateUserPreferencesResponse{}, nil
+	}
+
+	// Error if authenticated as anything other than a user
+	if claims.OwnerType() != auth.OwnerTypeUser {
+		return nil, fmt.Errorf("not authenticated as a user")
+	}
+
+	observability.AddRequestAttributes(ctx, attribute.String("user_id", claims.OwnerID()))
+	if req.Preferences.Timezone != nil {
+		observability.AddRequestAttributes(ctx, attribute.String("preferences_timezone", *req.Preferences.Timezone))
+	}
```

Comment:
Use `time.LoadLocation` (and import `_ "time/tzdata"`) to validate that it's a valid time zone